signing/verifier.go

```diff
@@ -0,0 +1,32 @@
+package signing
+
+import (
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/spacemeshos/ed25519"
+)
+
+type Verifier interface {
+	Verify(data []byte, sig []byte) (bool, error)
+	Bytes() []byte
+	String() string
+}
+
+type edVerifier struct {
+	pubKey []byte
+}
+
+func NewVerifier(pub []byte) (*edVerifier, error) {
+	return &edVerifier{pubKey: pub}, nil
```

Comment:
if you always return nil, why return an error?

Reply:
Removed